task/sort.go

```diff
@@ -38,7 +47,12 @@ func buildGraph(allTasks []Task, requiredTaskNames []string) ([]*graphNode, erro
 
 		if _, ok := seenTasks[task.Name()]; !ok {
 			seenTasks[task.Name()] = struct{}{}
-			g = append(g, &graphNode{task: task, edges: task.Dependencies()})
+			if err := validateFinallyTasks(allTasksMap, finallyTaskStates, task.Finally()); err != nil {
+				return nil, err
+			}
+			// toposort can
```

Comment:
unfinished comment?

Reply:
I thought this comment sufficiently explained the situation, but sure, happy to elaborate